nextcord/http.py

```diff
@@ -155,25 +154,255 @@ def bucket(self) -> str:
+        x_limit = response.headers.get("X-RateLimit-Limit")
+        self.limit = 1 if x_limit is None else int(x_limit)
+
+        # Updates the remaining left if it exists, being pessimistic.
+        x_remaining = response.headers.get("X-RateLimit-Remaining")
+
+        if x_remaining is None:
+            self.remaining = 1
+        elif self._first_update:
+            self.remaining = int(x_remaining)
+        else:
+            # If requests come back out of order, it's possible that we could get a wrong amount remaining.
+            # It's best to be pessimistic and assume it cannot go back up unless the reset task occurs.
+            self.remaining = (
+                int(x_remaining) if int(x_remaining) < self.remaining else self.remaining
+            )
+
+        # Updates the datetime of the reset.
+        x_reset = response.headers.get("X-RateLimit-Reset")
+        if x_reset is not None:
+            self.reset = datetime.utcfromtimestamp(float(x_reset))
+
+        # Updates the reset-after count, being pessimistic.
+        x_reset_after = response.headers.get("X-RateLimit-Reset-After")
+        if x_reset_after is not None:
+            x_reset_after = float(x_reset_after) + self._time_offset
+            if self.reset_after is None:
+                self.reset_after = x_reset_after
+            else:
+                self.reset_after = (
+                    x_reset_after if self.reset_after < x_reset_after else self.reset_after
+                )
+
+        # Updates the bucket name. The bucket name not existing as fine, as ``None`` is desired for that.
+        x_bucket = response.headers.get("X-RateLimit-Bucket")
+        self.bucket = x_bucket
+
+        if not self._reset_remaining_task or self._reset_remaining_task.done():
+            self.start_reset_task()
+
+        # If for whatever reason we have requests remaining but the reset event isn't set, set it.
+        if 0 < self.remaining and not self._on_reset_event.is_set():
+            _log.debug(
+                "Bucket %s updated with remaining %s, setting reset event.",
+                self.bucket,
+                self.remaining,
+            )
+            self._on_reset_event.set()
+
+        # If this is our first update, indicate that all future updates aren't the first.
+        if self._first_update:
+            self._first_update = False
+
+        _log.debug(
+            "Bucket %s updated with limit %s, remaining %s, reset %s, and reset_after %s seconds.",
+            self.bucket,
+            self.limit,
+            self.remaining,
+            self.reset,
+            self.reset_after,
+        )
+
+    def start_reset_task(self) -> None:
+        """Starts the reset task, non-blocking."""
+        loop = asyncio.get_running_loop()
+        _log.debug("Bucket %s will reset after %s seconds.", self.bucket, self.reset_after)
+        self._reset_remaining_task = loop.create_task(self.reset_remaining(self.reset_after))
```

Comment:
possibly want to check for existing tasks? might overwrite an already queued task is my thoughts

Reply:
I actually did this while fixing another issue I found, lol. Will push once I get some of the other issues I found worked out.